kafka/example_test.go

```diff
@@ -32,12 +32,11 @@ func ExampleSubscriber() {
 	ctx, cancel := context.WithTimeout(ctx, time.Second)
 	defer cancel()
 
-	// will block till context is cancelled or handler returns error:
+	// will block till context is cancelled:
 	err = sub.Topic("topic").Subscribe(
```

Comment:
so there is the other (more reliable IMO) option: we could do it the stan way and - instead of passing a context - return a subscription that can be stopped (deferred). the benefit of doing this is that calling `Stop()` should/would block until the handler exits. you wouldn't get the same guarantee with cancelling a context

Reply:
Yes, that's a goal for next PR. This one is just a prep work for that.